indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java

```diff
@@ -41,49 +44,61 @@ public class StacUtils {
 
     /**
      *
-     * @param polygons - Assume to be EPSG:4326 as this is what GeoJson use
+     * @param listOfPolygons - Assume to be EPSG:4326 as this is what GeoJson use
      * @return
      */
-    public static List<List<BigDecimal>> createStacBBox(List<Polygon> polygons) {
-        if(polygons != null) {
-            Envelope envelope = new Envelope();
-            try {
-                for (Polygon polygon : polygons) {
-                    // Add polygon one by one to extend the overall bounding box area, this is requirement
-                    // of STAC to have an overall bounding box of all smaller area as the first bbox in the list.
-                    if (polygon != null) {
-                        envelope.expandToInclude(polygon.getEnvelopeInternal());
-                    }
-                }
+    public static List<List<BigDecimal>> createStacBBox(List<List<Geometry>> listOfPolygons) {
+        List<List<BigDecimal>> result = new ArrayList<>();
+
+        if(listOfPolygons != null) {
+            final Envelope overallBoundingBox = new Envelope();
+            final AtomicBoolean hasBoundingBoxUpdate = new AtomicBoolean(false);
+            listOfPolygons
+                    .forEach(polygons -> {
+                        for (Geometry polygon : polygons) {
+                            // Add polygon one by one to expand the overall bounding box area, this is requirement
+                            // of STAC to have an overall bounding box of all smaller area as the first bbox in the list.
+                            if (polygon != null) {
+                                overallBoundingBox.expandToInclude(polygon.getEnvelopeInternal());
```

Comment:
should we check if `polygon.getEnvelopeInternal()` is null or empty first before flagging bounding box update to true?

Reply:
The list already check null before, but I can still add it